Links for Redactor

espen opened the pull request:
Includes the definedLinks Redactor plugin which does a request to a page which returns a list of defined links. Links are a list of all published pages on the site. The user can then easily make a link from one page to another without knowing the full path to the page.

GBH commented:
I like this a lot. Few comments:

Why the `position` on json objects. Why not just mimic nested structure?

```json
{
  name: 'foo',
  url: '/bar',
  children: [
    { ... },
    { ... }
  ]
}
```

- If you need to attach links path to js variable do it in `_head` view partial. No need to mangle file upload variable.
- Separate action for links, or index call with a parameter? I'm leaning more to index as it feels CRUDdy to me.
- `page.full_path` is no good. If Site has path, it won't give you right URL. You want to use `page.url(:relative)`
- if you're adding an entry to locale files, you gotta do it for all of them (even if untranslated). Super hard to figure out what's missing later.

This is definitely getting merged.

espen commented:
I added a JS variable, moved json to index, changed to relative url and updated locale files.

The plugin is from Redactor so I don't see why it should be prefixed with comfy. I haven't done any adjustments to the JS plugin from Redactor. There are more plugins here http://imperavi.com/redactor/plugins/ btw

GBH commented:
Didn't realize plugin already existed. Neat.
